### PR TITLE
fix farmer silent exit

### LINKED ISSUE #152: 🐞 [Bug]: Farmer quietly exits
### What happened?

More context there:
https://discord.com/channels/864285291518361610/1088790894196568126

### Version

0.1.11

### platform

- [X] linux
- [ ] windows
- [ ] macos

### Expected behavior

Plotting works

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ozgunozerk**
closes #152

### CODE REVIEW

**Review by i1i1 — COMMENTED** [review]
It looks weird. I think you just want to revert this change and make it work like this:
```rust
if is_verbose {
   signal::ctrl_c().await?;
} else {
   ....
}
```

**Comment by ozgunozerk** [comment]
No I didn't. 

In the `non-verbose` mode, we are waiting on the subscriptions.
Silent exit bug is happening on the `verbose` mode, since there is nothing being awaited on. 

So, as a psuedo-code:
```
if non-verbose:
    initialize and handle subscriptions

when kill signal received:
    if non-verbose:
        close subscriptions
    close farmer
    close node
```

**Comment by i1i1** [comment]
You right, makes sense. I would just prefer inlining those and making 2 different code paths. For example, this way, you won't even need to wrap `Farmer` with `Arc`.

**Comment by ozgunozerk** [comment]
`Arc` is not introduced with this PR. 
`Arc` is required because now `Farmer` is no longer `Clone`. And subscriptions do require `Farmer` as the argument. 
`Arc` was a simple enough way to get around this, while protecting safety.

**Comment by ozgunozerk** [comment]
that doesn't work since line 109 and 110 -> `plotting_subscribe_handle` and `solution_subscribe_handle` are not found. 
You are defining them in the `else` condition, yet trying to use them out of `else`

File: src/commands/farm.rs [hunk 1]
```diff
@@ -74,52 +75,71 @@ pub(crate) async fn farm(is_verbose: bool, executor: bool) -> Result<()> {
...
-        let handle = tokio::spawn(async move {
-            // if one of the subscriptions have not aborted yet, wait
-            // Plotting might end, so we ignore result here
-            let _ = plotting_subscribe_handle.await;
-            solution_subscribe_handle.await.expect_err("Solution subscription never ends");
+    // closing the subscriptions if there are any
+    if maybe_solution_handle.is_some() && maybe_plotting_handle.is_some() {
+        maybe_plotting_handle.as_ref().expect("check is above").abort();
+        maybe_solution_handle.as_ref().expect("check is above").abort();
+    }
```
Comment (i1i1) [inline src/commands/farm.rs:116]:
```suggestion
    if let Some((plotting_handle, solution_handle)) = maybe_handles.as_ref() {
        plotting_handle.abort();
        solution_handle.abort();
    }
```

**Review by i1i1 — APPROVED** [review]

**Pull request merged**